File: aspnet-in-practice/05/CS/DynamicData/ASPNET4InPractice.DynamicData/DynamicData/FieldTemplates/Phone.ascx.cs
using System;
using System.Collections.Specialized;
using System.ComponentModel.DataAnnotations;
using System.Web.DynamicData;
using System.Web;
using System.Web.UI;
using System.Web.UI.WebControls;

namespace ASPNET4InPractice.DynamicData
{
	public partial class PhoneField : System.Web.DynamicData.FieldTemplateUserControl
	{
		protected override void OnDataBinding(EventArgs e)
		{
			HyperLinkUrl.NavigateUrl = GetUrl(FieldValueString);
		}

		private string GetUrl(string phone)
		{
			return string.IsNullOrEmpty(phone) ? "#" : string.Concat("callto:", phone);
		}

		public override Control DataControl
		{
			get
			{
				return HyperLinkUrl;
			}
		}
	}
}

File: aspnet-in-practice/05/VB/DynamicData/ASPNET4InPractice.DynamicData/DynamicData/FieldTemplates/Phone.ascx.vb
Imports System
Imports System.Collections.Specialized
Imports System.ComponentModel.DataAnnotations
Imports System.Web.DynamicData
Imports System.Web
Imports System.Web.UI
Imports System.Web.UI.WebControls

Partial Public Class PhoneField
	Inherits System.Web.DynamicData.FieldTemplateUserControl
	Protected Overloads Overrides Sub OnDataBinding(ByVal e As EventArgs)
		HyperLinkUrl.NavigateUrl = GetUrl(FieldValueString)
	End Sub

	Private Function GetUrl(ByVal phone As String) As String
		Return If(String.IsNullOrEmpty(phone), "#", String.Concat("callto:", phone))
	End Function

	Public Overloads Overrides ReadOnly Property DataControl() As Control
		Get
			Return HyperLinkUrl
		End Get
	End Property
End Class